packages/chain-helpers/src/blockchain/Blockchain.ts

```diff
@@ -5,175 +5,160 @@
-    if (json instanceof Array) return json
-    return []
-  }
+export function parseSubscriptionOptions(
+  opts?: Partial<SubscriptionPromise.Options>
+): SubscriptionPromise.Options {
+  const {
+    resolveOn = IS_FINALIZED,
+    rejectOn = (result: ISubmittableResult) =>
+      EXTRINSIC_FAILED(result) || IS_ERROR(result),
+    timeout,
+  } = { ...opts }
 
-  public api: ApiPromise
-  private accountNonces: Map<IIdentity['address'], BN>
-
-  public constructor(api: ApiPromise) {
-    this.api = api
-    this.accountNonces = new Map<IIdentity['address'], BN>()
+  return {
+    resolveOn,
+    rejectOn,
+    timeout,
   }
+}
 
-  public async getStats(): Promise<BlockchainStats> {
-    const encoded: Text[] = await Promise.all([
-      this.api.rpc.system.chain(),
-      this.api.rpc.system.name(),
-      this.api.rpc.system.version(),
-    ])
-    const [chain, nodeName, nodeVersion] = encoded.map((el) => el.toString())
-    return { chain, nodeName, nodeVersion }
-  }
+/**
+ * [ASYNC] Submits a signed SubmittableExtrinsic and attaches a callback to monitor the inclusion status of the transaction
+ * and possible errors in the execution of extrinsics. Returns a promise to that end which by default resolves upon
+ * finalization and rejects any errors occur during submission or execution of extrinsics. This behavior can be adjusted via optional parameters.
+ *
+ * Transaction fees will apply whenever a transaction fee makes it into a block, even if extrinsics fail to execute correctly!
+ *
+ * @param tx The SubmittableExtrinsic to be submitted. Most transactions need to be signed, this must be done beforehand.
+ * @param opts Partial optional [[SubscriptionPromise]]to be parsed: Criteria for resolving/rejecting the promise.
+ * @returns A promise which can be used to track transaction status.
+ * If resolved, this promise returns ISubmittableResult that has led to its resolution.
+ */
+export async function submitSignedTx(
+  tx: SubmittableExtrinsic,
+  opts?: Partial<SubscriptionPromise.Options>
+): Promise<ISubmittableResult> {
+  log.info(`Submitting ${tx.method}`)
+  const options = parseSubscriptionOptions(opts)
+  const { promise, subscription } = makeSubscriptionPromise(options)
 
-  // TODO: implement unsubscribe as subscriptionId continuously increases
-  public async listenToBlocks(
-    listener: (header: Header) => void
-  ): Promise<() => void> {
-    return this.api.rpc.chain.subscribeNewHeads(listener)
-  }
+  let latestResult: SubmittableResult
+  const unsubscribe = await tx.send((result) => {
+    latestResult = result
+    subscription(result)
+  })
 
-  /**
-   * [ASYNC] Signs the SubmittableExtrinsic with the given identity or keyring pair.
-   *
-   * @param signer The [[IIdentity]] or KeyringPair to sign the tx with.
-   * @param tx The unsigned SubmittableExtrinsic.
-   * @param tip The amount of Femto-KILT to tip the validator.
-   * @returns Signed SubmittableExtrinsic.
-   */
-  public async signTx(
```

Comment:
Should we keep this as a function or do you think it is too trivial?

Reply:
(In case you want to sign and submit separately)

Reply:
Since getting a nonce is not necessary, this function will be replaced with a single call to tx.signAsync. I don’t see any need for it.